packages/flutter_tools/lib/src/build_system/targets/localizations.dart

```diff
@@ -36,6 +36,52 @@ Future<void> generateLocalizations({
     'bin',
     'gen_l10n.dart',
   );
+
+  // If generating a synthetic package, generate a warning if
+  // flutter: generate is not set.
+  if (options.useSyntheticPackage) {
+    final File pubspecFile = projectDir.childFile('pubspec.yaml');
```

Comment:
Use `var proj = FlutterProject.fromDirectory(environment.projectDir); flutterProject.manifest.generateSyntheticPackage` instead of parsing the pubspec yourself.